### ISSUE TITLE
When the input is larger make it as textarea

### ISSUE THREAD

**Issue by Paarmita** [publiclab/image-sequencer#389]
In the blend module, the input is the blend function, according to which blending should be done. 
So the input area shows function in a single line, it would look good if we can use the text area or some other component for it, instead of input.

![screenshot from 2018-10-06 01-34-24](https://user-images.githubusercontent.com/26546816/46557628-299ee280-c908-11e8-9600-64e32cd214ba.png)

**Comment by tech4GT** [comment]
@Paarmita Awesome! I was also thinking about this earlier today, maybe we can create a ui for this module by embedding a light weight text editor into it!

**Comment by tech4GT** [comment]
Since we expect people to write a complete function in there!

**Comment by Ankit-Singla** [comment]
@tech4GT I would like to work on this one. May I go ahead and start?

**Comment by tech4GT** [comment]
@Ankit-Singla Sure!! The one concern I have right now is that the editor needs to be as light weight as possible, so that it does not slow down the page!

**Comment by Dhiraj240** [comment]
@tech4GT i would like to work on this too.

**Comment by tech4GT** [comment]
@Dhiraj240 I think Ankit has already started on this one, maybe you guys can collaborate?

**Comment by Dhiraj240** [comment]
@Ankit-Singla  will you collaborate because i probably need help. Can we submit two different PRs for this same issue?

**Comment by Ankit-Singla** [comment]
@tech4GT awesome! I'll do this today. Are you okay with this @Dhiraj240 ?

**Comment by Dhiraj240** [comment]
Alright, as you wish i am here to learn @Ankit-Singla

**Comment by Ankit-Singla** [comment]
Cool @Dhiraj240 
I have created a branch named `inputtextarea' and have invited you to be a collaborator. Please accept the invitation so that we can work on this one together. 😄

**Comment by Dhiraj240** [comment]
@Ankit-Singla please remove your invitation as i am planning to give a direct PR so if you don't mind then i am working on it as already several days have passed without working.Let me give it a try and if i am not able to reproduce this issue, i will assign you.

**Comment by Ankit-Singla** [comment]
Sure @Dhiraj240 go ahead!

**Comment by Dhiraj240** [comment]
@Divy123 Sorry, i had assigned it to myself and will be sending PR soon.

**Comment by Divy123** [comment]
sure @tech4GT :smile:

**Comment by Dhiraj240** [comment]
@Divy123  you can work on this i am shifting to other projects and will just be reviewing PRs @tech4GT

**Comment by Divy123** [comment]
@Dhiraj240 sure I will work on this :smile:

**Comment by Divy123** [comment]
I googled about text editors that can be integrated and found these editors helpful :
* monaco-editor - https://www.npmjs.com/package/monaco-editor
* javascript-editor - https://www.npmjs.com/package/javascript-editor`
* Ace - https://ace.c9.io/
What would you suggest @tech4GT @Paarmita ?

**Comment by jywarren** [comment]
I wonder if we might decide to just show a simple `textarea` element --
could we have the info.json specify longer inputs by using "text" instead
of "string" in lines like this?

https://github.com/publiclab/image-sequencer/blob/89345f32b3dd2cf9f73fedbd03b9d371fbaf35ed/src/modules/ImportImage/info.json#L7

**Comment by Divy123** [comment]
Yeah that's cool to proceed like this. So should I proceed in this direction?

**Comment by Divy123** [comment]
Sure @HarshKhandeparkar but we are still on discussion. Also I think its not that same.

**Comment by HarshKhandeparkar** [comment]
@Divy123 what I meant by my comment was that we cannot change the type property directly from here. That has to be resolved first.

**Comment by Divy123** [comment]
Surely your idea is quite good at issue #570  but as I get since this issue is still under discussion, we are yet discussing about text-editor and text-area thing.

**Comment by Divy123** [comment]
No problem @HarshKhandeparkar . We all here to contribute so all the views are more than welcome.

**Comment by Harshithpabbati** [comment]
can i work on this??

**Comment by Divy123** [comment]
@Harshithpabbati 
I have already opened a PR for this. Please look into some other issue.
Thanks, we will be happy to get your contributions.
